src/js/jsx/shared/Dialog.jsx

```diff
@@ -324,8 +324,10 @@ define(function (require, exports, module) {
             if (this.props.dismissOnKeys && _.isArray(this.props.dismissOnKeys)) {
                 var flux = this.getFlux();
                 this.props.dismissOnKeys.forEach(function (keyObj) {
+                    var modifiers = keyObj.modifiers || {};
+
                     flux.actions.shortcuts.addShortcut(keyObj.key,
-                        keyObj.modifiers || {}, this.toggle, this.props.id + keyObj.key, true);
+                        modifiers, this.toggle, this.props.id + Object.keys(modifiers).join("+") + keyObj.key, true);
```

Comment:
The order of `Object.keys` can be a little hard to predict. Would it be similarly convenient and more reliable to use the modifier bits here instead?